pkg/flare/archive_security.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package flare
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mholt/archiver"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/status"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// CreateSecurityAgentArchive packages up the files
+func CreateSecurityAgentArchive(local bool, logFilePath string) (string, error) {
+	zipFilePath := getArchivePath()
+	return createSecurityAgentArchive(zipFilePath, local, logFilePath)
+}
+
+func createSecurityAgentArchive(zipFilePath string, local bool, logFilePath string) (string, error) {
+	b := make([]byte, 10)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", err
+	}
+
+	dirName := hex.EncodeToString(b)
+	tempDir, err := ioutil.TempDir("", dirName)
```

Comment:
Why generate a random `dirName` for `TempDir` ?

Reply:
Hm, this is borrowed from the regular flare, let me investigate if there is a reason.

Reply:
So it seems this is used in `createArchive` and `createDCAArchive` and now with this in `createSecurityArchive`, I can probably extract this into a helper. Hard to know if there are external assumptions somewhere as to how we create this dir. At least with this change it would need to be changed once.